File: dart-language/charge-system-demo/client-ng-dart/lib/src/history/history_component.dart
import 'package:angular/angular.dart';
import '../charge_service.dart';
import 'history_item_component.dart';

@Component(
    selector: 'my-history',
    styleUrls: ['history_component.css'],
    templateUrl: 'history_component.html',
    directives: [coreDirectives, HistoryItemComponent])
class HistoryComponent implements OnInit {
  final ChargeService chargeService;
  List<History> histories = [];

  HistoryComponent(this.chargeService);

  void ngOnInit() {
    this.getHistory();
  }

  void getHistory() async {
    final res = await chargeService.getHistory();
    histories = res.sublist(0, 4);
  }
}

File: dart-language/charge-system-demo/client-ng-ts/projects/at-mat/src/app/history/history.component.ts
import { Component, OnInit } from '@angular/core';
import { ChargeService, History } from '../charge.service';

@Component({
  selector: 'app-history',
  templateUrl: './history.component.html',
  styleUrls: ['./history.component.less']
})
export class HistoryComponent implements OnInit {
  histories: History[] = [];

  constructor(private chargeService: ChargeService) { }

  ngOnInit() {
    this.getHistory();
  }
  getHistory() {
    this.chargeService.getHistory().subscribe(res => {
      if (res) {
        this.histories = res;
      }
    });
  }

}
